taurus/entity/dict_serializable.py

```diff
@@ -82,7 +83,21 @@ def build(d):
         return loads(dumps(d))
 
     def __repr__(self):
-        return str(self.as_dict())
+        from taurus.util.impl import substitute_links, set_uuids
+        object_dict = self.as_dict()
+        # as_dict skips over keys in `skip`, but they should be in the representation.
+        skipped_keys = {x.lstrip('_') for x in vars(self) if x in self.skip}
+        for key in skipped_keys:
+            skipped_attribute = deepcopy(self.__getattribute__(key))
+            # Replace links in skipped keys with LinkByUID to prevent infinite recursion loop.
+            set_uuids(skipped_attribute)
+            if isinstance(skipped_attribute, list):
```

Comment:
Without this if statement, the `measurements` field in a `MaterialRun` will be a list of `LinkByUID`.